cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -1274,25 +1282,29 @@ def _deep_check(self):
         self._validate_seurat_convertibility()
 
         # Checks each component
-        for component, component_def in self.schema_def["components"].items():
-            logger.debug(f"Validating component: {component}")
+        for component_name, component_def in self.schema_def["components"].items():
+            logger.debug(f"Validating component: {component_name}")
+            component = getattr_anndata(self.adata, component_name)
+
             # Skip if component does not exist: only useful for adata.raw.var
-            if getattr_anndata(self.adata, component) is None:
+            if component is None:
+                # Check for required components
                 if "required" in component_def:
-                    self.errors.append(f"'{component}' is missing from adata and it's required.")
+                    pass
```

Comment:
I didn't even realize I skipped this. It goes to show that we have no test for this condition.
```suggestion
                    self.errors.append(f"'{component}' is missing from adata and it's required.")
```